lib/IDE/ArgumentCompletion.cpp

```diff
@@ -112,10 +164,19 @@ void ArgumentTypeCheckCompletionCallback::sawSolutionImpl(const Solution &S) {
   }
   auto ArgIdx = ArgInfo->completionIdx;
 
+  Type ExpectedCallType;
+  if (!isExpressionResultTypeUnconstrained(S, ParentCall)) {
+    ExpectedCallType = getTypeForCompletion(S, ParentCall);
+  }
+
   auto *CallLocator = CS.getConstraintLocator(ParentCall);
   auto *CalleeLocator = S.getCalleeLocator(CallLocator);
 
   auto Info = getSelectedOverloadInfo(S, CalleeLocator);
+  if (Info.Value && (Info.Value->shouldHideFromEditor() ||
+                     isInvalidInitRef(S, Info.Value, CallLocator))) {
```

Comment:
Could you please clarify why we need `isInvalidInitRef` instead of checking whether a particular kind of fix exists for the given location? Presumably if there was an invalid reference somewhere in the expression the solver would have recorded a fix before forming a solution?

Reply:
Good idea. Much simpler 👍